File: backend/src/main/java/com/smartcampus/backend/service/BookingService.java
package com.smartcampus.backend.service;

import com.smartcampus.backend.model.Booking;
import com.smartcampus.backend.model.BookingStatus;
import java.util.List;

public interface BookingService {
    Booking.ResponseDTO createBooking(Booking.RequestDTO dto);
    
    // Added for Frontend Update functionality
    Booking.ResponseDTO updateBooking(Long id, Booking.RequestDTO dto); 
    
    Booking.ResponseDTO getBookingById(Long id);
    List<Booking.ResponseDTO> getAllBookings();
    List<Booking.ResponseDTO> getBookingsByUser(Long userId);
    List<Booking.ResponseDTO> getBookingsByStatus(BookingStatus status);
    List<Booking.ResponseDTO> getBookingsByResource(Long resourceId);
    Booking.ResponseDTO approveBooking(Long id);
    Booking.ResponseDTO rejectBooking(Long id, String reason);
    Booking.ResponseDTO cancelBooking(Long id);
    void deleteBooking(Long id);
}

File: frontend/src/services/bookingService.js
import api from './api';

export const createBooking = (data) => api.post('/bookings', data);

export const updateBooking = (id, data) => api.put(`/bookings/${id}`, data);

export const getBookingById = (id) => api.get(`/bookings/${id}`);

export const getAllBookings = (status = null) => {
    const url = status ? `/bookings?status=${status}` : '/bookings';
    return api.get(url);
};

export const getUserBookings = (userId) => api.get(`/bookings/user/${userId}`);

export const getResourceBookings = (resourceId) => api.get(`/bookings/resource/${resourceId}`);

export const approveBooking = (id) => api.patch(`/bookings/${id}/approve`);

export const rejectBooking = (id, reason) => api.patch(`/bookings/${id}/reject`, { reason });

export const cancelBooking = (id) => api.patch(`/bookings/${id}/cancel`);

export const deleteBooking = (id) => api.delete(`/bookings/${id}`);
